aiidalab_widgets_base/process.py

```diff
@@ -55,19 +71,30 @@ def on_click(self, function):
 
     def on_btn_submit_press(self, _=None):
         """When submit button is pressed."""
-        self.submit_out.value = ''
+
+        if not self.append_output:
+            self.submit_out.value = ''
+
         input_dict = self.input_dictionary_function()
         if input_dict is None:
-            self.submit_out.value = "SubmitButtonWidget: did not recieve input dictionary."
+            if self.append_output:
+                self.submit_out.value += "SubmitButtonWidget: did not recieve input dictionary.<br>"
+            else:
+                self.submit_out.value = "SubmitButtonWidget: did not recieve input dictionary."
         else:
-            self.btn_submit.disabled = True
+            self.btn_submit.disabled = self.disable_after_submit
```

Comment:
Are there any other states in this widget that would cause the button to be disabled?

Reply:
No, I don't think so. It can only be disabled in the case of a successful submission, which is right here.

An additional possibility would be to have`process_class` set to `None` which is currently allowed, but shouldn't. I should make a check for that, I think.

Reply:
Ok, that's along the lines of what I was thinking. I think we must either disallow `None` or add the additional check here (something like this):
```suggestion
            self.btn_submit.disabled = self.process_node is None or self.disable_after_submit
```

Reply:
I was thinking of adding a check to `process_class`. One can just verify that it is a subclass of `Process`. This will automatically take care of not allowing the `process_class` to be `None`.

Reply:
done